sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricQueryResults.java

```diff
@@ -17,20 +17,29 @@
  */
 package org.apache.beam.sdk.metrics;
 
+import com.google.auto.value.AutoValue;
 import org.apache.beam.sdk.annotations.Experimental;
 import org.apache.beam.sdk.annotations.Experimental.Kind;
 
 /**
  * The results of a query for metrics. Allows accessing all of the metrics that matched the filter.
  */
+@AutoValue
 @Experimental(Kind.METRICS)
-public interface MetricQueryResults {
-  /** Return the metric results for the counters that matched the filter. */
-  Iterable<MetricResult<Long>> getCounters();
+public abstract class MetricQueryResults {
+  /** Return the metric results for the getCounters that matched the filter. */
```

Comment:
just noticed this `getCounters` comment typo (and similar below); I put a fix here in #7641 for now: https://github.com/apache/beam/pull/7641/commits/6a21a6bac2fe7e95afc345687961f8ef287b3042